scalding-hadoop-test/src/test/scala/com/twitter/scalding/platform/PlatformTest.scala

```diff
@@ -702,4 +723,21 @@ class PlatformTest extends WordSpec with Matchers with HadoopSharedPlatformTest
     }
   }
 
+  "An InvalidSourceTap that gets past validation" should {
+    "throw an InvalidSourceException" in {
+      val result = Try {
```

Comment:
well, it changes it from:

```
val result = Try {
  HadoopPlatformJobTest(new ReadPathJob(_), cluster)
    .arg("input", "/sploop/boop/doopity/doo/")
    .run
}

result match {
  case Failure(fe: FlowException) if Option(fe.getCause).exists(_.isInstanceOf[InvalidSourceException]) => ()
  case Failure(t) =>
    throw new RuntimeException("Expected InvalidSourceException wrapped in FlowException.", t)
  case _ =>
    throw new RuntimeException("Expected InvalidSourceException wrapped in FlowException, but the job succeeded")
}
```

to:

```
val result: FlowException = intercept[FlowException]
  HadoopPlatformJobTest(new ReadPathJob(_), cluster)
    .arg("input", "/sploop/boop/doopity/doo/")
    .run
}

assert(Option(result.getCause).exists(_.isInstanceOf[InvalidSourceException])
```

So, it saves you the case match and all the exception throwing and so on.
The intercept helper is itself an assertion that an exception should happen, so that's handled for you. I think it's fairly common to use, just as the `@Test(expected=FooException.class` junit annotation. Though intercept is much better because you get to inspect the exception.